spec/requests/graphql/mutations/digital_object/update_projects_spec.rb

```diff
@@ -0,0 +1,146 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Mutations::DigitalObject::UpdateProjects, type: :request do
+  include_context 'with stubbed search adapters'
+  let(:primary_project) { FactoryBot.create(:project) }
+  let(:secondary_project) { FactoryBot.create(:project) }
+  let(:tertiary_project) { FactoryBot.create(:project) }
+  let(:other_projects_set) { Set.new([secondary_project, tertiary_project]) }
+  let(:authorized_item) { FactoryBot.create(:item, primary_project: primary_project, other_projects: other_projects_set) }
+  let(:nonexistent_project_string_key) { "nonexistent_project" }
+
+  context 'when updating other_projects' do
+    let(:variables) do
+      {
+        input: {
+          id: authorized_item.uid,
+          primaryProject: {
+            stringKey: primary_project.string_key
+          },
+          otherProjects: [
+            { stringKey: tertiary_project.string_key }
+          ]
+        }
+      }
+    end
```

Comment:
Can we remove the primaryProject key-value pair from the variables to focus on only updating other_projects?

Reply:
The three cases (update primary, update other, update both) are now separately tested.